features/filesystem/bd/FlashSimBlockDevice.cpp

```diff
@@ -0,0 +1,149 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "FlashSimBlockDevice.h"
+#include "mbed_assert.h"
+#include <algorithm>
+#include <stdlib.h>
+#include <string.h>
+
+static const bd_size_t min_blank_buf_size = 32;
+
+static inline uint32_t align_up(bd_size_t val, bd_size_t size)
+{
+    return (((val - 1) / size) + 1) * size;
+}
+
+FlashSimBlockDevice::FlashSimBlockDevice(BlockDevice *bd, uint8_t erase_value) :
+    _erase_value(erase_value), _blank_buf_size(0),
+    _blank_buf(0), _bd(bd)
+{
+}
+
+FlashSimBlockDevice::~FlashSimBlockDevice()
+{
+    deinit();
+    delete[] _blank_buf;
+}
+
+int FlashSimBlockDevice::init()
+{
+    int ret = _bd->init();
+    if (ret) {
+        return ret;
+    }
+    _blank_buf_size = align_up(min_blank_buf_size, _bd->get_program_size());
+    if (!_blank_buf) {
+        _blank_buf = new uint8_t[_blank_buf_size];
+        MBED_ASSERT(_blank_buf);
```

Comment:
Non blocking and can be fixed in future - New will assert if memory allocation fails, additional check is not needed. Do you intent to use malloc instead here?

Reply:
Thanks, will remove the redundant check. And no - will keep using new[].